rust/noosphere-ipfs/src/client/kubo.rs

```diff
@@ -115,11 +124,16 @@ impl IpfsClient for KuboClient {
         let request_builder = Request::builder().method("POST").uri(&api_url.to_string());
         let request = form.set_body_convert::<Body, MultipartBody>(request_builder)?;
 
-        let response = self.client.request(request).await?;
-
-        match response.status() {
-            StatusCode::OK => Ok(()),
-            other_status => Err(anyhow!("Unexpected status code: {}", other_status)),
+        select! {
+            response = self.client.request(request) => {
+                match response?.status() {
+                    StatusCode::OK => Ok(()),
+                    other_status => Err(anyhow!("Unexpected status code: {}", other_status)),
+                }
+            },
+            _ = tokio::time::sleep(KUBO_DAG_IMPORT_TIMEOUT) => {
```

Comment:
`noosphere_core::wait`? the function could reasonably *not* be a "helper"

Reply:
If we did this, I would want to change the signature to take a `instant::Duration` or equivalent (to ensure the same granularity available via this API call).